### PR TITLE
Support Python3 in waterfall build scripts

### PR DESCRIPTION
**Pull request by dschuff**

### CODE REVIEW

File: src/parallel_runner.py [hunk 1]
```diff
@@ -14,8 +14,12 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
+from __future__ import print_function
 import multiprocessing
-import Queue
+try:
+  import queue
+except ImportError:
+  import Queue as queue
```
Comment (sbc100) [inline src/parallel_runner.py]: Comment here ?

File: src/host_toolchains.py [hunk 2]
```diff
@@ -66,7 +67,7 @@ def GetVSEnv(dir):
         continue
       var, val = e.split('=', 1)
       env[var] = val
-      print('ENV: %s = %s' % (var, val))
+      print(('ENV: %s = %s' % (var, val)))
```
Comment (sbc100) [inline src/host_toolchains.py]: Overzealous search and replace?

File: src/compile_torture_tests.py [hunk 3]
```diff
@@ -57,7 +57,7 @@ def find_runnable_tests(directory, pattern):
     for filename in files:
       if fnmatch.fnmatch(filename, pattern):
         fullname = os.path.join(root, filename)
-        with open(fullname, 'r') as f:
+        with open(fullname, 'r', encoding='ISO8859-1') as f:
```
Comment (sbc100) [inline src/compile_torture_tests.py]: wat?

File: src/build.py [hunk 4]
```diff
@@ -549,21 +552,26 @@ def SyncArchive(out_dir, name, url):
...
   work_dir = os.path.dirname(out_dir)
 
   try:
-    f = urllib2.urlopen(url)
-    print 'URL: %s' % f.geturl()
-    print 'Info: %s' % f.info()
+    if sys.version_info.major == 2:
+      from urllib2 import urlopen, URLError
+    else:
+      from urllib.request import urlopen, URLError
```
Comment (sbc100) [inline src/build.py]: Why if/else here but  try/except for `queue`

Maybe the if/else is better because it self documenting?

File: src/build.py [hunk 5]
```diff
@@ -125,7 +125,7 @@ def IsWindows():
 
 
 def IsLinux():
-  return sys.platform == 'linux2'
+  return sys.platform == 'linux2' or sys.platform == 'linux'
```
Comment (sbc100) [inline src/build.py]: `.startswith('linux')`?

File: src/compile_torture_tests.py (hunk 3 above)
Comment (dschuff) [reply src/compile_torture_tests.py]: Apparently it makes a difference. By default it tries UTF-8 encoding and fails to read several of the tests (the offending characters appear to be in comments)

Comment (sbc100) [reply src/compile_torture_tests.py]: Holy crap.   Can you add a command about this?

File: src/parallel_runner.py (hunk 1 above)
Comment (dschuff) [reply src/parallel_runner.py]: Replaced with explicit version check

File: src/build.py (hunk 4 above)
Comment (dschuff) [reply src/build.py]: Yeah I switched the other one for that reason.

**Pull request merged**